powerquery-docs/data-privacy-firewall.md

```diff
@@ -156,7 +156,7 @@ Here's a high-level summary of the partitioning logic.
     * Grouping (Static)
       * Partitions are merged, while maintaining separation between:
         * Partitions in different queries
-        * Partitions that reference other partitions vs. those that don't
+        * Partitions that reference multiple partitions vs. those that don't
```

Comment:
It's really "other partitions".

If the Source partition of QueryA references QueryB, then it will be treated as "a partition that references other partitions", since QueryB is a separate partition. Any partitions that refer only to this Source partition are treated the same and merged. Even if a downstream partition in QueryA refers to QueryC, that partition will be merged into the same partition as QueryA/Source.

If the Source partition of QueryA doesn't reference anything, then it will be treated as not referencing other partitions. Any partitions that refer only to this Source partition are treated the same and merged. But if a downstream partition in QueryA refers to QueryC, that partition won't be merged into the same partition as QueryA/Source.

Reply:
It's about preserving the separation between the two types of partitions in the *results* of grouping.

Reply:
Thanks, @ehrenMSFT.

I think a part that is throwing me the statement that "Partitions are merged, while maintaining separation between: [...] Partitions that reference other partitions vs. those that don't", which sounds like it applies _bidirectionally_. That is, two partitions cannot be merged if _either one_ references other partitions while the other does not. So `Employees/Merged Queries` and `Employees/Expanded Contacts` can't be merged because one reference other partitions while the other does not—but in actuality, they merge fine.

Is this another way of defining the logic that's used?

>  * Partitions can only be merged if both are in the same query.
>  * A partition will be merged into another if the only partition it references is the partition it is being merged into.

(So, only the partition that is _candidate_ for being merged in has to meet these rules [e.g. `Employees/Expanded Contacts`]. The partition receiving the merge does not [e.g. `Employee/Merged Queries`].)

Reply:
Yes, that's another way of describing the logic...almost. Partitions that both reference other queries, for example, can actually be merged.

Reply:
Is the latest commit any better? Thinking that something like this would help remove ambiguity in this particularly complex realm. :-)